src/com/owncloud/android/ui/activity/DrawerActivity.java

```diff
@@ -0,0 +1,702 @@
+     */
+    public abstract void allFilesOption();
+
+    /**
+     * Updates title bar and home buttons (state and icon).
+     * <p/>
+     * Assumes that navigation drawer is NOT visible.
+     */
+    protected void updateActionBarTitleAndHomeButton(OCFile chosenFile) {
+        super.updateActionBarTitleAndHomeButton(chosenFile);
+
+        /// set home button properties
+        if (mDrawerToggle != null) {
+            mDrawerToggle.setDrawerIndicatorEnabled(isRoot(chosenFile));
+        }
+    }
+
+    /**
+     * sets the given account name in the drawer in case the drawer is available. The account name is shortened
+     * beginning from the @-sign in the username.
+     *
+     * @param account the account to be set in the drawer
+     */
+    protected void setAccountInDrawer(Account account) {
+        if (mDrawerLayout != null && account != null) {
+            TextView username = (TextView) findNavigationViewChildById(R.id.drawer_username);
+            TextView usernameFull = (TextView) findNavigationViewChildById(R.id.drawer_username_full);
+            usernameFull.setText(account.name);
+            try {
+                OwnCloudAccount oca = new OwnCloudAccount(account, this);
+                username.setText(oca.getDisplayName());
+
+            } catch (Exception e) {
+                Log_OC.w(TAG, "Couldn't read display name of account; using account name instead");
+
+                username.setText(AccountUtils.getUsernameOfAccount(account.name));
+            }
+
+            DisplayUtils.setAvatar(account, (ImageView) findNavigationViewChildById(R.id.drawer_current_account),
+                    mCurrentAccountAvatarRadiusDimension, getResources(), getStorageManager());
+        }
+    }
+
+    /**
+     * fetches and sets the avatar of the current account in the drawer in case the drawer is available.
+     *
+     * @param account        the account to be set in the drawer
+     * @param menuItem       the menuItem to set the avatar on
+     */
+    private void setAvatar(Account account, MenuItem menuItem) {
+        if (mDrawerLayout != null && account != null) {
+
+            // Thumbnail in Cache?
+            Bitmap thumbnail = ThumbnailsCacheManager.getBitmapFromDiskCache("a_" + account.name);
+
+            if (thumbnail != null) {
+                menuItem.setIcon(
+                        BitmapUtils.bitmapToCircularBitmapDrawable(MainApp.getAppContext().getResources(), thumbnail)
+                );
+            } else {
+                // generate new avatar
+                if (ThumbnailsCacheManager.cancelPotentialAvatarWork(account.name, menuItem)) {
+                    final ThumbnailsCacheManager.AvatarGenerationTask task =
+                            new ThumbnailsCacheManager.AvatarGenerationTask(
+                                    menuItem, getStorageManager(), account
+                            );
+                    if (thumbnail == null) {
```

Comment:
This block has some problems. This condition always will be ```true``` at this point, and the block inside should try to create the retrieve the avatar that doesn't exist yet, right? It doesn't, it just generates the colored letter. 

Please, @AndyScherzinger, tell me if I'm losing something.